scipy/special/cephes/bdtr.c

```diff
@@ -148,103 +148,92 @@
 
 #include "mconf.h"
 
-double bdtrc(k, n, p)
-int k, n;
-double p;
+double bdtrc(double k, int n, double p)
 {
     double dk, dn;
 
     if (npy_isnan(p)) {
 	return NPY_NAN;
     }
-    if ((p < 0.0) || (p > 1.0)) {
-	goto domerr;
+    if (p < 0.0 || p > 1.0 || n < floor(k)) {
```

Comment:
Hm the `n < floor(k)` feels weird (versus `n < k`).

For the future (i.e. not this PR) we could think about extending the domain of the CDF for `-oo < k < oo`, and similarly with the other survival functions/CDFs.

Reply:
I'll infer from your comment here that for now I'll leave this as it is currently written. 

If we continue to proceed with interpretation as a cdf then the values for `k` can be `0` for `k<0` and `1` for `k>n`.